ClientApp/viking-vault/src/Common/AdminIcon.tsx

```diff
@@ -0,0 +1,109 @@
+import React from "react";
+import icon from "../Resources/images/user.png";
+import "./styles.css";
+import {Redirect} from "react-router-dom";
+import { constants } from "../Resources/Constants";
+import UserIconAvatar from "./UserIconAvatar"
+
+const API_URL = `${constants.baseUrl}userprofilepages`;
+
+interface IAdminIconState {
+  clicked: boolean;
+  redirect: boolean;
+  adminImage: string;
+}
+
+class AdminIcon extends React.Component<any, IAdminIconState> {
+  megaMenu: any;
+
+  constructor(props: any) {
+    super(props);
+    this.state = {
+      clicked: false,
+      redirect: false,
+      adminImage: icon
+    };
+    this.megaMenu = React.createRef();
+    this.handleOutsideClick = this.handleOutsideClick.bind(this);
+  }
+
+  sendData = () =>{
+    setTimeout(() => {
+      this.props.parentCallBack(false);
+    }, 100);
```

Comment:
What does this `100` represents?

Reply:
It is a timeout that allows for a state changing of a referred component before rendering another component